orderfulfiller/orderfulfiller.go

```diff
@@ -106,26 +107,53 @@ func (r *OrderFulfiller) startOrderTimeoutWorker(ctx context.Context) {
 				lmt.Logger(ctx).Error("error getting expired orders", zap.Error(err))
 				continue
 			}
+
 			for _, order := range orders {
-				if fulfillmentStatus, err := r.fillHandler.UpdateFulfillmentStatus(ctx, order); err != nil {
+				fulfillmentStatus, err := r.fillHandler.UpdateFulfillmentStatus(ctx, order)
+				if err != nil {
 					lmt.Logger(ctx).Warn(
 						"error updating fulfillment status",
 						zap.Error(err),
 						zap.String("orderID", order.OrderID),
 						zap.String("sourceChainID", order.SourceChainID),
+						zap.String("destinationChainID", order.DestinationChainID),
 					)
-				} else if fulfillmentStatus == dbtypes.OrderStatusExpiredPendingRefund && r.shouldRefundOrders {
-					if err := r.fillHandler.InitiateTimeout(ctx, order); err != nil {
-						lmt.Logger(ctx).Warn(
-							"error initiating timeout",
-							zap.Error(err),
-							zap.String("orderID", order.OrderID),
-							zap.String("sourceChainID", order.SourceChainID),
-						)
-					} else {
-						lmt.Logger(ctx).Info("successfully initiated timeout", zap.String("orderID", order.OrderID), zap.String("sourceChainID", order.SourceChainID))
-					}
+					continue
+				}
+
+				// do not try and refund this order
+				if !r.shouldRefundOrders || fulfillmentStatus != dbtypes.OrderStatusExpiredPendingRefund {
```

Comment:
if an order has status OrderStatusExpiredPendingRefund and solver has r.shouldRefundOrders set to false, it will be processed everytime this worker loop is run. is the intention here to keep it that way so if the solver turns shouldRefundOrders on it retroactively refunds them? or should this set status to abandoned/another db type like OrderStatusExpiredNotRefunded?

Reply:
ok this is a little confusing, but the order timeout worker (this function) actually won't even be running if `shouldRefundOrders` is false. the `Run` function for the order fulfiller looks like this
```go
func (r *OrderFulfiller) Run(ctx context.Context) {
	if r.shouldRefundOrders {
		go r.startOrderTimeoutWorker(ctx)
	}
	if r.shouldFillOrders {
		go r.startOrderFillWorkers(ctx)
		r.dispatchOrderFills(ctx)
	}
}
```

Reply:
yeah i missed that, this isnt an issue then 👍 thanks for clarifying